### PR TITLE
refactor: replace null by undefined

### PR DESCRIPTION
**Pull request by nasreddineskandrani**
## Description
following this: https://medium.com/@nasreddine.skandrani/front-end-typescript-null-vs-undefined-51900143c6a6

I changed `null` to `undefined` to give it a try as discussed.

### CODE REVIEW

**Comment by aminpaks** [comment]
I think you should chill out about some stuff in React world. Not the same rules apply to React as they do in Angular with TypeScript.

File: src/Header/Header.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const Header: FC<HeaderProps> = ({
     <CollapsePadding variant="left-s-collapse">
       <div>
         <CollapsePadding variant="left-l-collapse">
-          {logo != null && <StyledNavLogoContainer>{logo}</StyledNavLogoContainer>}
+          {logo != undefined && <StyledNavLogoContainer>{logo}</StyledNavLogoContainer>}
```
Comment (aminpaks) [inline src/Header/Header.tsx]: This is literally the same and actually `logo != null` is shorter!

File: src/UI-Kits/Grid/Grid.tsx [hunk 2]
```diff
@@ -15,12 +15,12 @@ export const Grid: FC<PGridProps> & { Col: typeof Column } = ({
...
   // Calculate all the children's sizes in an array
-  const allSizes: (number | null)[] = [];
+  const allSizes: (number | undefined)[] = [];
   React.Children.forEach(children, child => {
     if (React.isValidElement<PGridColumnProps>(child)) {
-      allSizes.push(child.props.size || null);
+      allSizes.push(child.props.size || undefined);
     } else {
-      allSizes.push(null);
+      allSizes.push(undefined);
```
Comment (aminpaks) [inline src/UI-Kits/Grid/Grid.tsx]: This can introduce a bug potentially.

File: src/Header/Header.tsx (hunk 1 above)
Comment (nasreddineskandrani) [reply src/Header/Header.tsx]: i know i just changed it with replace of IDE we should use `===` i ll correct it.

**Comment by nasreddineskandrani** [comment]
basically if we read this article also:
https://basarat.gitbooks.io/typescript/docs/javascript/null-undefined.html
In the conclusion he say:

> Final thoughts
TypeScript team doesn't use null : TypeScript coding guidelines and it hasn't caused any problems. Douglas Crockford thinks null is a bad idea and we should all just use undefined.
However, NodeJS style code bases uses null for Error arguments as standard as it denotes Something is currently unavailable. I personally don't care to distinguish between the two as most projects use libraries with differing opinions and just rule out both with == null.


i found 2 cases:
1. find a problem
https://developer.mozilla.org/en-US/docs/Web/API/Headers/get (edited) 
the get of the header is using `null`
so the `undefined` won't work here
2. `undefined` is used here: https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/find

So from this i do agree with the guy that wrote the article on top:
> I personally don't care to distinguish between the two as most projects use libraries with differing opinions and just rule out both with == null.

And i don't agree with
> Douglas Crockford thinks null is a bad idea and we should all just use undefined.
since the community is using the 2.

From this i have to say that i was wrong  thinking it's better to use only one.

**Comment by nasreddineskandrani** [comment]
another recent interesting exchange about the topic in Apollo context:
https://github.com/apollographql/apollo-tooling/issues/622

**Pull request closed without merging**